File: Ruby/class_object_hero.rb
class Hero
  @@attack_types = {
    warrior: 'espada',
    mage: 'magia',
    monk: 'artes maciais',
    ninja: 'shuriken'
  }

  def initialize(name, age, hero_type)
    @name = name
    @age = age
    @hero_type = hero_type
    @attack_type = @@attack_types[hero_type]
  end

  def attack
    puts "O #{@hero_type} atacou usando #{@attack_type}."
  end
end

hero1 = Hero.new('Zeno', 351, :ninja)
hero2 = Hero.new('Zero', 539, :mage)

hero1.attack
hero2.attack

File: Rust/class_object_hero.rs
use std::collections::HashMap;

struct Hero {
    name: String,
    age: u32,
    hero_type: &'static str,
    attack_type: &'static str,
}

impl Hero {
    fn new(name: &str, age: u32, hero_type: &'static str, attack_types: &HashMap<&'static str, &'static str>) -> Hero {
        let attack_type = attack_types.get(hero_type).unwrap_or(&"desconhecido");
        Hero { name: name.to_string(), age, hero_type, attack_type }
    }

    fn attack(&self) {
        println!("O {} atacou usando {}.", self.hero_type, self.attack_type);
    }
}

fn main() {
    let attack_types: HashMap<&'static str, &'static str> = [
        ("warrior", "espada"),
        ("mage", "magia"),
        ("monk", "artes marciais"),
        ("ninja", "shuriken")
    ].iter().cloned().collect();

    let hero1 = Hero::new("Zeno", 523, "ninja", &attack_types);
    let hero2 = Hero::new("Zero", 649, "mage", &attack_types);

    hero1.attack();
    hero2.attack();
}
